Add metadata to standalone examples

As a follow-up to #11540, we would like to add metadata to as many examples as possible. This will not only make the examples more usable as they are right now, but it will also open up new possibilities to search for and crosslink examples.

For the purpose of this issue, 'standalone examples' are all .py files in these folders in this repository:

* `examples/plotting`
* `examples/models`

## Prerequisites 

Adding metadata uses reStructuredText (ReST), like the rest of Bokeh's documentation. However, most of the work can probably be done without knowing much about ReST. To test the metadata, you need to build the docs locally. In order to do that, you first need to [set up a local development environment](https://docs.bokeh.org/en/latest/docs/dev_guide/setup.html), and then follow the instructions for [setting up and building Bokeh’s documentation](https://docs.bokeh.org/en/latest/docs/dev_guide/documentation.html#setting-up-and-building-bokeh-s-documentation).

## Adding metadata

The **metadata to add to the examples** are added as a multi-line strings comment at the beginning of each .py file, containing the following information:
* An intro sentence or two describing the example narratively (please add line breaks to keep line length to a max of 80 chars)
* Entries for a custom-made Sphinx directive called `bokeh-example-metadata` with the following data
    * `:sampledata:` the name of the sample data set, if applicable. This is the list of the available sets of sample data: https://docs.bokeh.org/en/latest/docs/reference/sampledata.html?highlight=sampledata#module-bokeh.sampledata. For example: `:sampledata: glucose` 
    * `:apis:` The paths to the API objects highlighted in this example. Can be one or several. For example `:apis: bokeh.layouts.column, bokeh.layouts.gridplot`. Please use the specific model, i.e. `bokeh.models.Circle` instead of `bokeh.models`.
    *  `:refs:` Sphinx references to the user guide that cover the functionalities highlighted in this example. This part uses breadcrumbs, i.e. `chapter > sub-section`. To find the name of the references, you need to check the ReST source code of each of the respective user guide chapters in the `sphinx/source/docs/user_guide/` directory of the repository. A reference is the name (with underscores) between `.. _` and `:` above a headline. For example: ``:refs: :ref:`userguide_plotting` > :ref:`userguide_plotting_scatter_markers`, :ref:`userguide_data` > :ref:`userguide_data_transforming``
    * `:keywords:` A list of comma-separated keywords to help finding this example. For example: :keywords: alpha, colormap, markermap, scatter

## Example

See https://docs.bokeh.org/en/latest/docs/gallery/marker_map.html (and its source https://github.com/bokeh/bokeh/blob/branch-3.0/examples/plotting/file/marker_map.py) for a complete example. This is the ReST source code at the beginning of the Python file:

```
''' A scatter plot using the `Palmer penguin dataset`_. This example
demonstrates color and marker mapping with basic plot elements. The chart
shows correlation between body mass and flipper length for three different
penguin species.

.. bokeh-example-metadata::
    :sampledata: penguins
    :apis: bokeh.plotting.Figure.scatter, bokeh.transform.linear_cmap, bokeh.transform.factor_mark
    :refs: :ref:`userguide_plotting` > :ref:`userguide_plotting_scatter_markers`, :ref:`userguide_data` > :ref:`userguide_data_transforming`
    :keywords: alpha, colormap, markermap, scatter

.. _Palmer penguin dataset: https://github.com/allisonhorst/palmerpenguins
'''
```
This creates the following result:
![image](https://user-images.githubusercontent.com/39711796/138810127-0fda116d-87ce-43d7-8828-aaa83368ada0.png)

hey , i am trying to add metadata to examples/plotting/custom_tooltips.py but i am running into a problem with the package exclusion tests . I have checked the files where the supposed import happens but i can't see the imports . 
An example is this: 
```
__ test_no_typing_extensions_common_individual[bokeh.sphinxext.bokeh_sitemap] __

module = 'bokeh.sphinxext.bokeh_sitemap'

    @pytest.mark.parametrize('module', MODULES)
    def test_no_typing_extensions_common_individual(module) -> None:
        proc = run([python, "-c", verify_clean_imports('typing_extensions', [module])])
>       assert proc.returncode == 0, f"typing_extensions imported in common module {module}"
E       AssertionError: typing_extensions imported in common module bokeh.sphinxext.bokeh_sitemap
E       assert 1 == 0
E        +  where 1 = CompletedProcess(args=['/home/wmwine/.pyenv/versions/3.10.6/envs/kivu/bin/python3.10', '-c', "import sys; import bokeh.sphinxext.bokeh_sitemap; sys.exit(1 if any('typing_extensions' in x for x in sys.modules.keys()) else 0)"], returncode=1).returncode
```
but  the file src/bokeh/sphinxext/bokeh_sitemap.py does not have any typing_extensions imports   
Thanks alot for the help

--- a/examples/models/colors.py
+++ b/examples/models/colors.py
@@ -1,5 +1,5 @@
 '''This example demonstrates the usage of grouping different shades
-of same color together using bokeh groups.
+of the same color together using Bokeh groups.
 
 A hover tooltip displays the URL of the color to go to.
 

--- a/examples/models/colors.py
+++ b/examples/models/colors.py
@@ -1,3 +1,16 @@
+'''This example demonstrates the usage of grouping different shades
+of same color together using bokeh groups.
+
+A hover tooltip displays the URL of the color to go to.
+
+Navigate to the specific color by tapping.
+
+.. bokeh-example-metadata::
+    :apis: bokeh.models.axes.CategoricalAxis, bokeh.models.tools.HoverTool, bokeh.models.plots.Plot
+    :refs: :ref:`ug_interaction_tools_hover_tool`
+    :keywords: groups, hover, color, tap, tooltip
+
+'''
 from math import pi
 
 import pandas as pd

